pkg/reconciler/taskrun/resources/taskref_test.go

```diff
@@ -448,14 +451,6 @@ func TestGetTaskFunc_RemoteResolution(t *testing.T) {
 			taskYAMLString,
 		}, "\n"),
 		wantTask: parse.MustParseV1Task(t, taskYAMLString),
-	}, {
```

Comment:
Curious why do we remove these tests?

Reply:
The webhook will now validate that beta features aren't used in v1 tasks unless enable-api-fields is set to beta. This replaces the existing hard-coded check for this specific bit of validation (https://github.com/tektoncd/pipeline/blob/080ea1367aa342915fe86c8da583ee3401983946/pkg/reconciler/taskrun/resources/taskref.go#L173-L177). I could mock the tests to return a BadRequestError specifically when beta features are used in v1 crds with enable-api-fields set to "stable", but I think the newly added tests that return BadRequestError are sufficient coverage and less brittle.